### PR TITLE
Settings functions

### PR DESCRIPTION
**Pull request by Tavernoshca**

### CODE REVIEW

File: crashbin_app/api/urls.py [hunk 1]
```diff
@@ -8,5 +8,6 @@
 
 urlpatterns = [
     path('', include(router.urls)),
-    path('report/new/', views.ReportNew.as_view())
+    path('report/new/', views.ReportNew.as_view()),
+    path('bins/<int:pk>/subscribe', views.bin_subscribe, name='bin_subscribe')
```
Comment (The-Compiler) [inline crashbin_app/api/urls.py]: Add a trailing comma:

```suggestion
    path('bins/<int:pk>/subscribe', views.bin_subscribe, name='bin_subscribe'),
```

that way, future additions only have a one-line diff :wink:

File: crashbin_app/api/views.py [hunk 2]
```diff
@@ -19,3 +23,14 @@ def post(self, request, *args, **kwargs):
 class BinViewSet(viewsets.ModelViewSet):
     queryset = Bin.objects.all()
     serializer_class = BinSerializer
+
+
+@api_view(['POST'])
+def bin_subscribe(request: HttpRequest, pk: int) -> HttpResponse:
+    user = request.user  # type: ignore
+    bin_obj: Bin = Bin.objects.get(id=pk)
+    if user in bin_obj.subscribers.all():
```
Comment (The-Compiler) [inline crashbin_app/api/views.py:32]: Is the `.all()` needed?

File: crashbin_app/templates/crashbin_app/bin_detail.html [hunk 3]
```diff
@@ -100,8 +123,15 @@ <h5 class="m-0 mr-2">Github Issue</h5>
                         <div class="d-flex flex-row mb-1 header-underline">
                             <h5 class="m-0 mr-2">Actions</h5>
                         </div>
-                        <a class="btn btn-primary my-1" href="#" role="button">Subscribe</a>
-                        <a class="btn btn-success btn-sm my-1" href="#" role="button">Archive</a>
+                        <button type="button" id="buttonSubscribe"
+                                class="btn btn-{% if user in bin.subscribers.all %}outline-{% endif %}primary my-1">
```
Comment (The-Compiler) [inline crashbin_app/templates/crashbin_app/bin_detail.html:127]: Is the `.all` needed?

File: crashbin_app/templates/crashbin_app/report_detail.html [hunk 4]
```diff
@@ -158,25 +159,5 @@ <h5 class="m-0 mr-2">Actions</h5>
...
-                    Are you sure you want to delete this report?
-                </div>
-                <div class="modal-footer">
-                    <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
-                    <button type="button" id="modalDeleteButton" class="btn btn-danger">Delete report</button>
-                </div>
-            </div>
-        </div>
-    </div>
+    {% modal_delete None %}
```
Comment (The-Compiler) [inline crashbin_app/templates/crashbin_app/report_detail.html]:
```suggestion
    {% modal_delete %}
```

**Review by The-Compiler — CHANGES_REQUESTED** [review]

**Review by The-Compiler — APPROVED** [review]

File: crashbin_app/api/views.py (hunk 2 above)
Comment (Tavernoshca) [reply crashbin_app/api/views.py:32]: Unfortunately yes

**Pull request merged**